tests/acceptance/standard/test_hooks_processor.py

```diff
@@ -0,0 +1,149 @@
+import logging
+import pytest
+import textwrap
+import time
+from typing import TYPE_CHECKING
+
+import gitlab
+from gitlab.v4.objects import ProjectHook, Project, Group
+
+from gitlabform import GitLabForm
+from gitlabform.gitlab import GitlabWrapper
+from gitlabform.gitlab.core import NotFoundException
+
+from tests.acceptance import allowed_codes, run_gitlabform, create_group, create_project
+
+logger = logging.getLogger(__name__)
+
+
+def create_test_hook(project: Project):
+    hook = project.hooks.create(
+        {
+            "url": "http://birdman.chirp/update",
+            "push_events": False,
+            "merge_requests_events": False,
+        }
+    )
+    if TYPE_CHECKING:
+        assert isinstance(hook, ProjectHook)
+    return hook
+
+
+def prepare_project(gl, group: str, project: str):
+    with allowed_codes(404):
+        gl.groups.delete(group)
+        # wait for delete to finish
+        time.sleep(6)
+    created_group: Group = create_group(group)
+    created_project: Project = create_project(created_group, project)
+    return
+
+
+def reset_gitlab(gl):
+    with allowed_codes(404):
+        gl.groups.delete("test_group")
+        time.sleep(5)
+    return
+
+
+@pytest.fixture
+def test_yaml():
+    return """
+      projects_and_groups:
+        "*":
+          project_settings:
+            visibility: internal
+        test_group/mystery:
+          project_settings:
+            rick: pickle
+          hooks:
+            "http://plumbus.org/create":
+                push_events: true
+                merge_requests_events: true
+            "http://birdman.chirp/update":
+                push_events: true
+      """
```

Comment:
If you use the existing fixtures for setting up projects, you won't need the full config or hard-code the project name here. Just the `hooks` config should be enough. Then you can move this into the relevant test method. It'd be easier to read/debug the config that is being tested if this is within the test method.